### PR TITLE
DB changes for task template and task instance addition

### PR DESCRIPTION
**Pull request by sushanthakumar**
**What this PR does / why we need it**:
This PR contains the changes for addition of task template to db

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
Dev Test Report: Added test code to check add and get for task template table
image

**Release note**:

```release-note
```

### CODE REVIEW

File: delfin/db/api.py [hunk 1]
```diff
@@ -498,3 +498,50 @@ def alert_source_get_all(context, marker=None, limit=None, sort_keys=None,
     """
     return IMPL.alert_source_get_all(context, marker, limit, sort_keys,
                                      sort_dirs, filters, offset)
+
+
```
Comment (ThisIsClark) [inline delfin/db/api.py:502]: Do we need the task update api?

File: delfin/db/api.py [hunk 2]
```diff
@@ -498,3 +498,50 @@ def alert_source_get_all(context, marker=None, limit=None, sort_keys=None,
...
+def task_template_get(context, task_template_id):
+    """Get a task_template or raise an exception if it does not exist."""
+    return IMPL.task_template_get(context, task_template_id)
+
+
+def task_template_get_all(context, marker=None, limit=None, sort_keys=None,
+                          sort_dirs=None, filters=None, offset=None):
+    """Retrieves all  task templates.
```
Comment (ThisIsClark) [inline delfin/db/api.py:520]: Duplicated space, and please to check other place for duplicated spaces

File: delfin/db/api.py [hunk 3]
```diff
@@ -498,3 +498,50 @@ def alert_source_get_all(context, marker=None, limit=None, sort_keys=None,
...
+
+
+def task_template_delete_by_storage(context, storage_id):
+    """Delete all task_templates of given storage or raise an exception if it
+    does not exist. """
+    return IMPL.task_template_delete_by_storage(context, storage_id)
+
+
+def task_template_delete(context, task_template_id):
+    """Delete a given task template or raise an exception if it does not
```
Comment (ThisIsClark) [inline delfin/db/api.py:549]: And from python specifications, when comments more than one line, the tail `"""` should be put in a new line
https://www.python.org/dev/peps/pep-0008/#documentation-strings
Same for other comments

File: delfin/db/api.py (hunk 1 above)
Comment (sushanthakumar) [reply delfin/db/api.py:502]: Not needed currently, but added for future use

File: delfin/db/api.py (hunk 2 above)
Comment (sushanthakumar) [reply delfin/db/api.py:520]: Updated

File: delfin/db/api.py (hunk 3 above)
Comment (sushanthakumar) [reply delfin/db/api.py:549]: Updated

**Comment by ThisIsClark** [comment]
LGTM

**Review by ThisIsClark — APPROVED** [review]
LGTM

**Review by NajmudheenCT — APPROVED** [review]
Overall LGTM.
some improvement points
Hope the tests can be be improved.. fakes are empty in most cases, every time it will return a success, This is applicable to overall DP API tests, but lets start with new models..

**Pull request merged**